lockfile/src/parse_depfile.rs

```diff
@@ -94,10 +99,9 @@ pub fn parse_depfile(
         }
     }
 
-    // Generate lockfile if path might be a manifest and feature and option are
-    // enabled.
+    // Generate lockfile for likely manifests when feature and option are enabled.
     #[cfg(feature = "generator")]
-    if let Some(generation_path) = _generation_path.filter(|_| maybe_manifest) {
+    if let Some(generation_path) = _generation_path.filter(|_| manifest_likely) {
         return Ok(generate_lockfile(&generation_path, &path, format, parser)?);
     }
```

Comment:
This can now call `generate_lockfile()` with a `format` that doesn't support lockfile generation... For example:

```
> ./target/debug/phylum parse -t msbuild Cargo.toml
❗ Error: unsupported manifest file "Cargo.toml"

❗ Error: could not parse dependency file "Cargo.toml"

Caused by:
    Dependency file parsing failed
```

Of course, that's a pretty unusual case... But I think it would be slightly better to return the error that we've stored from attempting to parse it as a lockfile (i.e., `lockfile_error`).

Reply:
I _think_ I was able to make the requested change. I'm not confident it was done in the most Rusty way so please feel free to provide better implementations.